### DISCUSSION TITLE
Problem with the Refresh Token - API return states it is invalid

### DISCUSSION THREAD

**Discussion by Terry-JF** [firefly-iii/firefly-iii#9113] [Firefly III]
I have FF3 self-installed on Linux Mint and working well. (Thanks James). The auto-import is fetching transactions from multiple banks via GoCardless every day, triggered by cron. I also have curl accessing the FF3 API from a Windows system to pull these transactions across to another system and keeping these in-sync by deleting superceeded 'pending' transactions as 'booked' versions appear, as well as syncing categories.

Once I realised that the access token expires, I followed the details provided by M-Paladin in post #4595 and all worked well up until the point where my script uses curl to try to refresh the tokens:

curl --request POST \
  --url 'http://192.168.4.163:88/oauth/token' \
  --header 'content-type: application/x-www-form-urlencoded' \
  --data grant_type=refresh_token \
  --data 'client_id=3' \
  --data client_secret=<password_oauth_client as shown in /profile> \
  --data refresh_token=<a_long_hexadecimal_refresh_token which I have previously saved in database>

And I get back:

{
	"error":"invalid_request",
	"error_description":"The refresh token is invalid.",
	"hint":"Cannot decrypt the refresh token",
	"message":"The refresh token is invalid."
}

I have been through the loop of creating new access_token/refresh_token pairs 3 times and being careful to copy the entire refresh_token. I note that the refresh_token is a bit shorter than the access_token and looks to be pure hexadecimal. Although the access_tokens work fine the refesh_token is rejected as per the above every time. Can you see anything wrong with my curl code or is there something about the way the refresh tokens work that I am missing? I note that #4595 is from 2021 so has anything changed in the way the refresh tokens work since that time?

Many thanks, Terry

**Comment by JC5** [comment]
On my end nothing has changed, but I must admit I never used it like this.
